kernel/src/actions/mod.rs

```diff
@@ -141,6 +141,7 @@ pub struct Add {
     pub path: String,
 
     /// A map from partition column to value for this logical file.
+    #[schema_container_values_null]
```

Comment:
We should consider `//` commenting why the attribute is needed? 
(if I understand correctly, partition values can be physically null, and the attribute causes the read to correctly handle that even tho the resulting HashMap will simply not contain them)

However, the above explanation is a bit hard to parse, and doesn't immediately follow from the attribute. That makes me wonder if we need to take a different approach to avoid trouble. 

For example -- should the attribute be called `drop_null_container_values`? Then we could document it as "I know the underlying data for this non-nullable map/vec is nullable -- drop null values instead of erroring out"